MIP/mip-58/README.md

```diff
@@ -0,0 +1,462 @@
+    /// Anyone can initiate a bridge transfer from the source chain  
+    /// The amount is burnt from the initiator and the module-level nonce is incremented  
+    /// @param initiator The initiator's Ethereum address as a vector of bytes.  
+    /// @param recipient The address of the recipient on the Aptos blockchain.  
+    /// @param amount The amount of assets to be locked.  
+    public entry fun initiate_bridge_transfer(  
+        initiator: &signer,  
+        recipient: vector<u8>,  
+        amount: u64  
+    ) {  
+        let initiator_address = signer::address_of(initiator);  
+        let ethereum_address = ethereum::ethereum_address(recipient);  
+    
+        // Increment and retrieve the nonce  
+        let nonce = increment_and_get_nonce(initiator_address);  
+    
+        // Create bridge transfer details  
+        let details = atomic_bridge_store::create_details(  
+            initiator_address,  
+            ethereum_address, 
+            amount,  
+            nonce  
+        );  
+    
+        // Generate a unique bridge transfer ID  
+        let bridge_transfer_id = bridge_transfer_id(&details);  
+    
+        // Add the transfer details to storage  
+        atomic_bridge_store::add(bridge_transfer_id, details);  
+    
+        // Burn the amount from the initiator  
+        atomic_bridge::burn(initiator_address, amount);  
+    
+        // Emit an event with nonce  
+        event::emit(  
+            BridgeTransferInitiatedEvent {  
+                bridge_transfer_id,  
+                initiator: initiator_address,  
+                recipient,  
+                amount,  
+                nonce,  
+            }  
+        );  
+    }  
+
+    /// Completes a bridge transfer by the initiator.  
+    ///  
+    /// @param caller The signer representing the bridge operator.  
+    /// @param initiator The initiator's Ethereum address as a vector of bytes.  
+    /// @param bridge_transfer_id The unique identifier for the bridge transfer.  
+    /// @param recipient The address of the recipient on the Aptos blockchain.  
+    /// @param amount The amount of assets to be locked.  
+    /// @param nonce The unique nonce for the transfer.  
+    
+    /// @abort If the caller is not the bridge operator.  
+    public entry fun complete_bridge_transfer(  
+        caller: &signer,  
+        bridge_transfer_id: vector<u8>,
+        initiator: vector<u8>,  
+        recipient: address,  
+        amount: u64,  
+        nonce: u64  
+    ) {  
+        atomic_bridge_configuration::assert_is_caller_operator(caller);  
+        let ethereum_address = ethereum::ethereum_address(initiator);
+        let details = atomic_bridge_store::create_details(  
+            ethereum_address,  
+            recipient,  
+            amount,  
+            nonce  
+        );
+
+        // asserts that parameters are correct  
+        assert!(bridge_transfer_id == bridge_transfer_id(&details), 0x1);
```

Comment:
I don't really understand the check here. 

1. The `initiator` contract on L1 is trusted: the events with the data `bridgeTransferId, initiator, recipient, amount, nonce` are not corrupted and final (assuming we relay only from final blocks). So the `bridgeTransfer` in the event is equal to the `keccak` of the other parameters
2. The relayer is trusted, so it only relays `bridgeTransferId, initiator, recipient, amount, nonce` without tampering with them. 
3. in `complete_bridge_transfer()` (line 387)  we verify that the caller is the operator, which is the relayer (should be).
4. The `create_details` (line 387)  creates a bridge transfer id by hashing the input param (same as keccak) and only checks that this `transferId` has not been completed before.
5. the `assert!` at line 405 should never fail.

All the data `bridgeTransferId, initiator, recipient, amount, nonce` are published on L1 (emitted in an event). 
If I know  `initiator, recipient, amount, nonce`, I can easily compute the `transferId` (line 218, `keccak`).

What added protection does line 405 provide?